data/state/peerAccount.go

```diff
@@ -0,0 +1,369 @@
+	entry, err := NewPeerJournalEntryJailTime(a, a.JailTime)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.JailTime = jailTime
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// SetCurrentShardIdWithJournal sets the account's shard id, saving the old state before changing
+func (a *PeerAccount) SetCurrentShardIdWithJournal(shId uint32) error {
+	entry, err := NewPeerJournalEntryCurrentShardId(a, a.CurrentShardId)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.CurrentShardId = shId
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// SetNextShardIdWithJournal sets the account's shard id, saving the old state before changing
+func (a *PeerAccount) SetNextShardIdWithJournal(shId uint32) error {
+	entry, err := NewPeerJournalEntryNextShardId(a, a.NextShardId)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.NextShardId = shId
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// SetNodeInWaitingListWithJournal sets the account's nodes status whether in waiting list, saving the old state before
+func (a *PeerAccount) SetNodeInWaitingListWithJournal(nodeInWaitingList bool) error {
+	entry, err := NewPeerJournalEntryInWaitingList(a, a.NodeInWaitingList)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.NodeInWaitingList = nodeInWaitingList
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// IncreaseValidatorSuccessRateWithJournal increases the account's number of successful signing,
+// saving the old state before changing
+func (a *PeerAccount) IncreaseValidatorSuccessRateWithJournal() error {
+	entry, err := NewPeerJournalEntryValidatorSuccessRate(a, a.ValidatorSuccessRate)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.ValidatorSuccessRate.NrSuccess++
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// DecreaseValidatorSuccessRateWithJournal increases the account's number of missed signing,
+// saving the old state before changing
+func (a *PeerAccount) DecreaseValidatorSuccessRateWithJournal() error {
+	entry, err := NewPeerJournalEntryValidatorSuccessRate(a, a.ValidatorSuccessRate)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.ValidatorSuccessRate.NrFailure++
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// IncreaseLeaderSuccessRateWithJournal increases the account's number of successful signing,
+// saving the old state before changing
+func (a *PeerAccount) IncreaseLeaderSuccessRateWithJournal() error {
+	entry, err := NewPeerJournalEntryLeaderSuccessRate(a, a.LeaderSuccessRate)
+	if err != nil {
+		return err
+	}
+
+	a.accountTracker.Journalize(entry)
+	a.LeaderSuccessRate.NrSuccess++
+
+	return a.accountTracker.SaveAccount(a)
+}
+
+// DecreaseLeaderSuccessRateWithJournal increases the account's number of missing signing,
+// saving the old state before changing
+func (a *PeerAccount) DecreaseLeaderSuccessRateWithJournal() error {
```

Comment:
IncreaseLeaderFailureRateWithJournal instead DecreaseLeaderSuccessRateWithJournal ?

Reply:
the fields name is successRate, I wanted to mask what happens in the back.